File: 2D_sym_python/plants/Guarana.py
from Plant import*

class Guarana(Plant):

    def __init__(self):
        super().__init__()
        self.name='u'
    
    def collision(self, world, organismToFight):
        if self.isAttacker==True:
            return True
        else:
            if LETTER_RANGE_ANIMAL_A < ord(organismToFight.GetName()) <LETTER_RANGE_ANIMAL_Z:
                tempStrength = organismToFight.GetStrength()*3
                organismToFight.SetOrganismStrength(tempStrength)
                world.AddCommentatorVector(reportType.STRENGTH_INC,organismToFight,None,0)
        return False
    
    def GetName(self):
        return 'u'
    
    def GetOrganismFullName(self):
        return "Guarana"

File: 2DVirtualWorldSimulatorCpp/2DVirtualWorldSimulator/Guarana.cpp
#include <iostream>
#include "Guarana.h"
#include "World.h"

Guarana::Guarana() {
	this->name = 'u';
}

bool Guarana::collision(World* world,Organism* organismToFight) {
	int tempStrength;
	if (this->isAttacker == true) {
		return true;
	}
	else {
		//Strength of the animal which ate guarana is permanently increased by 3.
		if (organismToFight->GetName() > LETTER_RANGE_ANIMAL_A && organismToFight->GetName() < LETTER_RANGE_ANIMAL_Z) {
			tempStrength = (organismToFight->GetStrength()) * 3;
			organismToFight->SetOrganismStrength(tempStrength);
			world->addCommentatorVector(STRENGTH_INC, organismToFight, nullptr, NULL);
		}
	}
	return false;
}

char Guarana::GetName() const {
	return 'u';
}

string Guarana::GetOrganismFullName() const {
	return "Guarana";
}

Guarana::~Guarana() {

}